src/ansys/fluent/core/workflow.py

```diff
@@ -356,37 +357,43 @@ def python_name(self) -> str:
             Pythonic name of the task.
         """
         if not self._python_name:
-            display_name_map = self._command_source._help_string_display_text_map
-            if self.display_name() not in display_name_map.values():
-                try:
-                    this_command = self._command()
-                    # temp reuse helpString
-                    self._python_name = camel_to_snake_case(
-                        this_command.get_attr("helpString")
+            if self._command_source._dynamic_python_names:
+                display_name_map = self._command_source._python_name_display_text_map
+                if self.display_name() not in display_name_map.values():
+                    self._set_python_name()
+                else:
+                    self._python_name = get_first_dict_key_for_value(
+                        display_name_map, self.display_name()
                     )
-                    if (
-                        self._python_name
-                        in self._command_source._help_string_display_text_map
-                    ):
-                        self._populate_duplicate_task_list()
-                    else:
-                        self._command_source._help_string_display_text_map[
-                            self._python_name
-                        ] = self.display_name()
-                    self._command_source._help_string_command_id_map[
-                        self._python_name
-                    ] = this_command.command
-                    self._command_source._help_string_display_id_map[
-                        self._python_name
-                    ] = this_command.get_attr("displayText")
-                except Exception:
-                    pass
             else:
-                self._python_name = list(display_name_map.keys())[
-                    list(display_name_map.values()).index(self.display_name())
-                ]
+                self._set_python_name()
+
         return self._python_name
 
+    def _set_python_name(self):
+        try:
+            this_command = self._command()
+            # temp reuse helpString
+            self._python_name = camel_to_snake_case(this_command.get_attr("helpString"))
+            self._cache_data(this_command)
+        except Exception:
+            pass
+
+    def _cache_data(self, command):
+        _disp_text = command.get_attr("displayText")
```

Comment:
Minor: generally we don't prefix `_` to local variable names

Reply:
Done, Thank you.